contracts/FragToken.sol

```diff
@@ -49,6 +49,13 @@ contract FRAGToken is ERC20, ERC20Permit, Ownable {
 
     function lock(uint256 amount, bytes calldata signature, Period period) external {
         require(amount > 0, "Amount must be greater than 0");
+        require(
```

Comment:
gas wise this is probably bad as you then repeat the same logic under

Reply:
Right, they could just be in one single revert.